common-jvm/src/main/java/org/ergoplatform/ErgoFacade.kt

```diff
@@ -123,21 +125,18 @@ fun sendErgoTx(
     try {
         val ergoClient = getRestErgoClient(prefs)
         return ergoClient.execute { ctx: BlockchainContext ->
-            val proverBuilder = ctx.newProverBuilder()
-                .withMnemonic(
-                    signingSecrets.mnemonic,
-                    signingSecrets.password,
-                    // TODO BIP-32 fix signingSecrets.deprecatedDerivation
-                )
-            derivedKeyIndices.forEach {
-                proverBuilder.withEip3Secret(it)
-            }
-            val prover = proverBuilder.build()
+            val prover = buildProver(ctx, signingSecrets, derivedKeyIndices)
 
             val contract: ErgoContract = recipient.toErgoContract()
-            val unsignedTx = BoxOperations.createForEip3Prover(prover, ctx).withAmountToSpend(amountToSend)
-                .withInputBoxesLoader(ExplorerAndPoolUnspentBoxesLoader().withAllowChainedTx(true))
-                .withTokensToSpend(tokensToSend).putToContractTxUnsigned(contract)
+            val unsignedTx =
+                BoxOperations.createForEip3Prover(prover, ctx).withAmountToSpend(amountToSend)
+                    .withFeeAmount(feeAmount)
+                    .withInputBoxesLoader(
+                        ExplorerAndPoolUnspentBoxesLoader().withAllowChainedTx(
+                            true
+                        )
+                    )
```

Comment:
Yes, this looks better. The automatic code formatter sometimes overacts and it comes to different results based on the code it is working on. So sometimes it can be corrected, in this case it can. Sometimes it can't, when the line is too long.